lib/march_hare/exception_handler/rubify_java_exception_handler.rb

```diff
@@ -0,0 +1,31 @@
+require 'set'
+module MarchHare
+  module RubifyJavaExceptionHandler
+    def initialize(logger)
+      super()
+      @logger = logger
+    end
+
+    def log(msg, error)
+      if is_socket_closed_or_connection_reset(error)
+        logger.warn(msg)
+        logger.warn(error)
+      else
+        logger.error(msg)
+        logger.error(error)
+      end
+    end
+
+    attr_reader :logger
+
+    SOCKET_CLOSE_OR_CONNECTION_RESET_MESSAGE = Set.new([
+      "Connection reset",
+      "Socket closed",
+      "Connection reset by peer"
+    ])
+
+    def is_socket_closed_or_connection_reset(error)
```

Comment:
I had to re-implement this here as it's private in the java method and therefore not accessible in subclasses
https://github.com/rabbitmq/rabbitmq-java-client/blob/master/src/main/java/com/rabbitmq/client/impl/ForgivingExceptionHandler.java#L126